python/cairo-ec/src/cairo_ec/circuits/ec_ops.cairo

```diff
@@ -22,19 +22,55 @@ func ec_double(x0: felt, y0: felt, a: felt) -> (felt, felt) {
     end:
 }
 
-// @dev Assert that a point is, or is not, on the curve by checking that either y is actually the square root of rhs
-//      (is_on_curve = True => y^2 = rhs) or y is the square root of rhs * g (is_on_curve = False => y^2 = rhs * g),
-//      which mean that rhs is not a quadratic residue because g * rhs is, and so that x is not on the curve.
+// @notice Verifies that a point (x, y) satisfies a specific elliptic curve condition based on is_on_curve.
+// @dev If is_on_curve = 1, asserts y^2 = x^3 + ax + b, confirming the point is on the curve.
+//      If is_on_curve = 0, asserts y^2 = g * (x^3 + ax + b), verifying that no `y` satisfies the curve equation for the given `x`.
+//      Use this when you need to prove a point meets one of these equations, not to fail on invalid points.
+//      For explicit rejection of on-curve points, see assert_not_on_curve.
 // @param x The x coordinate of the point
 // @param y The y coordinate of the point
-// @param g The generator point
-// @param is_on_curve True if the point is on the curve, False otherwise
-func assert_is_on_curve(x: felt, y: felt, a: felt, b: felt, g: felt, is_on_curve: felt) {
-    assert is_on_curve * (1 - is_on_curve) = 0;
+// @param a The a coefficient of the curve
+// @param b The b coefficient of the curve
+// @param g A scalar used when is_on_curve = 0
+// @param is_on_curve Flag (0 or 1) indicating which condition to verify
+func assert_x_is_on_curve(x: felt, y: felt, a: felt, b: felt, g: felt, is_on_curve: felt) {
```

Comment:
I fell this function should be 2 separate functions.
g is not used when is_on_curve = 0 but can still be passed. 
Out of scope of the PR I suppose

Reply:
The purpose is that you want to ensure that either
- there exists an `y` s.t. (x,y) is on curve
- or there is no such `y`, in which case, you verify `y^2 = g * (x^3 + ax + b)`, verifying that no `y` satisfies the curve equation for the given `x`.

Reply:
I agree couldn't this be done in 2 distincts functions?
It would ease the comprehension and review. 
note: this is a nit and out of scope